e2etest/clean/clean_distributor.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package main
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"log"
+	"strconv"
+)
+
+func main() {
+	//set up aws go sdk config
+	testSession, err := session.NewSession()
+	if err != nil {
+		log.Fatalf("Error creating session %v", err)
+	}
+	// set up aws go sdk ssm client
+	ssmClient := ssm.New(testSession)
+
+	//get list of documents with versions
+	documentName := "testAWSDistroOTel-Collector"
+	listDocumentVersionInput := ssm.ListDocumentVersionsInput{Name: &documentName}
+	listDocumentVersionsOutput, err := ssmClient.ListDocumentVersions(&listDocumentVersionInput)
+	if err != nil {
+		log.Fatalf("Error getting document list %v", err)
+	}
+	versions := make([]*string, 0, len(listDocumentVersionsOutput.DocumentVersions))
```

Comment:
Maybe make a struct to hold both the string and int values so you don't have to convert back and forth.

```go
type documentTuple struct {
	str string
	num int
}
```
Then, you would just add all of them to the versions slice and only need to do two loops. One to get the largest `documentTuple` and append them to the slice and one to set the default/delete all the others.